lib/class-wp-theme-json-gutenberg.php

```diff
@@ -315,11 +320,14 @@ private static function sanitize( $input, $valid_block_names, $valid_element_nam
 		}
 		$schema_styles_blocks   = array();
 		$schema_settings_blocks = array();
+
 		foreach ( $valid_block_names as $block ) {
-			$schema_settings_blocks[ $block ]           = self::VALID_SETTINGS;
+			$schema_settings_blocks[ $block ]           = self::VALID_SETTINGS[ $block ] ? array_merge( self::VALID_SETTINGS, self::VALID_SETTINGS[ $block ] ) : self::VALID_SETTINGS;
```

Comment:
Here if we have block level settings we merge with the top level settings to preserve backwards compat but allow extension at a block level.